### ISSUE TITLE
RKE2 air-gap install fails - /install-cni.sh\": stat /install-cni.sh: no such file or directory: unknown

### ISSUE THREAD

**Issue by schadam14** [rancher/rke2#1554]
**Environmental Info:**
RKE2 Version: 
rke2 version v1.20.8+rke2r1 

Node(s) CPU architecture, OS, and Version: 
Linux 4.18.0-193.el8.x86_64 #1 SMP Fri May 8 10:59:10 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

Cluster Configuration: 
1st master node

**Describe the bug:**
First time, I installed rke2 with the following command: 
sudo curl -sfL https://get.rke2.io | sh -
(Everything was working fine.)

Then I wanted to reinstall it using the air-gap install method:
- I ran the rke2-killall.sh, then rke2-uninstall.sh
- Restarted the host
- Copied rke2 binary to /usr/bin, and rke2-images.linux-amd64.tar.gz to /var/lib/rancher/rke2/agent/images/
- I created the initial /etc/rancher/rke2/config.yaml configuration file
- - with configured private-registry and system-default-registry
- Started rke2 server

**Steps To Reproduce:**

- Install RKE2 air-gap, using tarball method
- I set the following flags in the config.yaml:

token: my-token		
tls-san:			
  - hostname
  - IP address

etcd-snapshot-schedule-cron: '* */6 * * *'
etcd-snapshot-retention: 4

private-registry: /etc/rancher/rke2/registries.yaml
system-default-registry: "my-private-registry"

secrets-encryption: true

kube-apiserver-arg:
   - tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_RSA_WITH_AES_256_GCM_SHA384
   - enable-admission-plugins=NodeRestriction,PodSecurityPolicy,AlwaysPullImages

kubelet-arg:
   - tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_RSA_WITH_AES_256_GCM_SHA384
   - event-qps=0

**Expected behavior:**
Pods starting successfully.

**Actual behavior:**
NAMESPACE     NAME                                        READY   STATUS                  RESTARTS   AGE
kube-system   etcd-d-rke2sbsrv-01                         1/1     Running                 0          13m
kube-system   helm-install-rke2-canal-gmp2j               0/1     Completed               0          14m
kube-system   helm-install-rke2-coredns-m6l2f             0/1     Completed               0          14m
kube-system   helm-install-rke2-ingress-nginx-gjb4k       0/1     Pending                 0          14m
kube-system   helm-install-rke2-kube-proxy-fr6cz          0/1     Completed               0          14m
kube-system   helm-install-rke2-metrics-server-ncr9m      0/1     Pending                 0          14m
kube-system   kube-apiserver-d-rke2sbsrv-01               1/1     Running                 0          14m
kube-system   kube-controller-manager-hostname      1/1     Running                 0          14m
kube-system   kube-proxy-497cb                            0/1     CrashLoopBackOff        7          13m
kube-system   kube-scheduler-d-rke2sbsrv-01               1/1     Running                 0          14m
kube-system   rke2-canal-fzv9g                            0/2     Init:CrashLoopBackOff   7          13m
kube-system   rke2-coredns-rke2-coredns-6597db677-94cgn   0/1     Pending                 0          13m

**Additional context / logs:**

 rke2-canal-fzv9g:
"StartContainer for \"d3909838292e279797893db006bab40ac8246cfd7f1c8a1bbf2b95f8108a2bba\" failed" error="failed to create containerd task: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: \"/install-cni.sh\": stat /install-cni.sh: no such file or directory: unknown"

kube-proxy-497cb:
StartContainer for \"af0da1914d6490d9c35c0ccbe68c11601f95251bcc1855091ae04f9024e575ca\" failed" error="failed to create containerd task: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: \"/usr/local/bin/kube-proxy\": stat /usr/local/bin/kube-proxy: no such file or directory: unknown"

**Comment by brandond** [comment]
What OS distribution? Is it possible you're missing selinux policy RPMs or something else that's was uninstalled, but not reinstalled due to not running the install script?

**Comment by schadam14** [comment]
It seems the issue was caused by the corrupted images loaded into my private repository. As I understand, not every image is loaded from the tarball at the startup, if the image is also represented at the system-default-registry. For the first 2 attempt I used the rancher-save-images.sh and rancher-load-images.sh to load the rke2 images into my private repository. (Manually corrected the path in the images.txt before pushing the images.) Somehow it is not working well with the rke2 images, I had to manually pull, tag and push the images.  After that I dropped the tarball and successfully created the cluster from my private registry.
